### ISSUE TITLE
Extra spaces in html tagged templates

### ISSUE THREAD

**Issue by DiogoDoreto** [prettier/prettier#7096]
**Prettier 1.19.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEAbAhgIzm0wF5MALGAW2wAMAeNAB1ygD4ASYeADxgF96Aemas2tANwAdKCAA0ICExgBLdMlC4ATlogB3AAraEaZCFwA3CCoAm8kPi24wAazgwAyizAqoAc2QYLQBXOAVKGgB1chV4UTA4DxNYlQtYgE8zMDRTBV80OC0YAyc-KlxkADNcbAKFACs0HgAhJ1d3D1wqOAAZXzgqmrqQRp4PXz9sOABFYIh4QdqwkBYtAq0zfAIieyYtXxhI2xhyZAAOAAYFPYgCyKcmMz24dYsBhQBHOfgSpVMUXBoAC0UDgcBs4PsWjgXxU0JKuDKFSQ1SWCgKVBUgRCyzQEyms3mAxRQ2WMAIRxsJ2QACYFEFcCpsBMAMIQKjlMxQaDvEDBAoAFQI-1RwwsoQAklAIbAPGB9soAILSjwwdJTRYFfj8IA)
```sh
--parser babel
```

**Input:**
```jsx
const label = html`<span>${text}</span>`;

```

**Output:**
```jsx
const label = html`
  <span>${text}</span>
`;

```

**Expected behavior:**

Adding whitespace inside the tagged template literal affects the output. To avoid the spaces around my components, the only way right now is to add ignore comments.

**Comment by saschanaz** [comment]
@evilebottnawi For me #7101 wasn't exactly this issue though, because hyperHTML automatically trim out the newlines *if single-rooted* and thus won't make a breaking change.

**Comment by saschanaz** [comment]
What should be done here? Any formatting outside a tag is potentially breaking. I think we should introduce an option to:

* turn off the whole HTML formatting
* turn off formatting outside of any tags
* or, not insert newlines before and after the root elements.

**Comment by DiogoDoreto** [comment]
@saschanaz I'd vote to keep it simple: if the user tries to have it as one-liner and the line is not too long, avoid adding the line breaks. Otherwise leave the default formatting behavior.

This will create an incentive to the user to have to simplify their content, possibly extracting it into simple variables that would be easier to read.

It then will behave in a similar way as this example:

```js
// the line below is left as-is
const obj = [{ test: true }];

// the code below will be reformatted with line breaks
// input:
const obj = [{ 
  test: true }];
// output:
const obj = [
  {
    test: true
  }
];
```

What do you think?

**Comment by saschanaz** [comment]
That doesn't even need an option then! Sounds good to me 👍

**Comment by saschanaz** [comment]
@evilebottnawi Do prettier maintainers think https://github.com/prettier/prettier/issues/7096#issuecomment-564444006 is good? Is it possible to implement that?

**Comment by LarsDenBakker** [comment]
I would really like to see this as well.

**Comment by evilebottnawi** [comment]
@saschanaz it is out of scope this bug, we should not add extra space in this case, so PR welcome should be not hard

**Comment by evilebottnawi** [comment]
Fixed #7208
